openmrs-19/src/main/java/org/motechproject/openmrs19/service/impl/OpenMRSPersonServiceImpl.java

```diff
@@ -93,7 +93,20 @@ public Person createPerson(String configName, String givenName, String familyNam
     public Person updatePerson(String configName, Person person) {
         try {
             Config config = configService.getConfigByName(configName);
-            Person updated = personResource.updatePerson(config, person);
+            Person fetchedPerson = personResource.getPersonById(config, person.getUuid());
+
+            Person.Address addressForUpdate = fetchedPerson.getPreferredAddress();
+            if(addressForUpdate != null) {
+                person.getPreferredAddress().setUuid(addressForUpdate.getUuid());
+            }
+            personResource.updatePersonAddress(config, person.getUuid(), person.getPreferredAddress());
+
+            Person.Name nameForUpdate = fetchedPerson.getPreferredName();
+            person.getPreferredName().setUuid(nameForUpdate.getUuid());
+            personResource.updatePersonName(config, person.getUuid(), person.getPreferredName());
+
+            Person generalInformationForUpdate = prepareGeneralInformationForUpdate(person);

```

Comment:
As far as I understand, the name and address values should not be sent in update person request. The better/more readable way to do this is using PersonUpdateSerializer